src/ExecuteTimeWidget.ts

```diff
@@ -346,6 +354,7 @@ export default class ExecuteTimeWidget extends Widget {
       .composite as boolean;
     this._settings.historyCount = settings.get('historyCount')
       .composite as number;
+    this._settings.dateFormat = settings.get('dateFormat').composite as string;
```

Comment:
With the current state:
- user can still set invalid value from setting form (it should not be possible)
- if settings are invalid the error is still silent (not shown to the user) - users should not need to know about existence of developer console
- because the error is handled late, it logs in console for every cell executed

I would suggest that try-catch validation would happen when settings are set (this is on the line to which this comment is anchored), allowing fallback to the default value, and that the regex is moved to the schema to allow real-time validation when user is changing settings.

```suggestion
    const dateFormat = settings.get('dateFormat').composite as string;
    const formatValidationResult = validateDateFormat(dateFormat);
    if (formatValidationResult.isValid) {
      this._settings.dateFormat = dateFormat;
    } else {
      // fallback to default
      this._settings.dateFormat = 'yyy-MM-dd HH:mm:ss';
      // warn user once
      showErrorMessage(
        'Invalid date format',
        formatValidationResult.message
      );
    }
```

Reply:
For reference, [`showErrorMessage`](https://jupyterlab.readthedocs.io/en/latest/api/functions/apputils.showErrorMessage.html) needs to be imported from `@jupyterlab/apputils`.

Reply:
That is a good point. Let me implement it